File: SERVER_PERCORSO/Mex.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Bus_Percorsi;

namespace Mex
{
    public class Base
    {
        public bool Status;
        public List<string> Error;
    }
    public class ServerIMG 
    {
        public class Request
        {
            public string img;
        }
        public class Response : Base
        {
            public Fermata fermata;
            public List<Fermata> ArriviProb;
        }
    }
    public class ServerBuses
    {
        public class Request
        {
            public Fermata Partenza;
            public Fermata Arrivo;

        }
        public class Response : Base
        {
            public List<Bus> buses;
        }
    }
    public class ServerNearBus 
    {
        public class Request
        {
            public List<Bus> buses;
        }
        public class Response : Base
        {
            public Bus bus;
        }
    }
    public class ServerPosition 
    {
        public class Request
        {
            public Bus bus;
        }
        public class Response : Base
        {
            public string img;
        }
    }
}

File: Flutter/app/lib/models/mex.dart
import 'bus.dart';
import 'fermata.dart';

class Base {
  late bool status;
  late List<String> error;
}

class RequestImg {
  String img;
  RequestImg(this.img);
}

class ResponseImg implements Base {
  late bool status;
  late List<String> error;
  late Fermata fermata;
  late List<Fermata> arriviprob;
}

class RequestBus {
  late Fermata partenza;
  late Fermata arrivo;
}

class ResponseBus implements Base {
  late bool status;
  late List<String> error;
  late Fermata attuale;
  late List<Bus> buses;
}

class RequestNearBus {
  late Fermata attuale;
  late List<Bus> buses;
}

class ResponseNearBus implements Base {
  late bool status;
  late List<String> error;
  late Bus bus;
}

class RequestPosition {
  late Bus bus;
}

class ResponsePosition implements Base {
  late bool status;
  late List<String> error;
  late String img;
}
